feat: refactor to use the latest auro-button

jason-capsule42 opened the pull request:
# Alaska Airlines Pull Request

- BREAKING CHANGE: API refactored to match the latest auro-button version
- Resolves: AlaskaAirlines/auro-cli#186  AlaskaAirlines/auro-backtotop#66

## Checklist:

- [x] My update follows the CONTRIBUTING guidelines of this project
- [x] I have performed a self-review of my own update

**By submitting this Pull Request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

_Pull Requests will be evaluated by their quality of update and whether it is consistent with the goals and values of this project. Any submission is to be considered a conversation between the submitter and the maintainers of this project and may require changes to your submission._

**Thank you for your submission!**<br>
-- Auro Design System Team

## Summary by Sourcery

Refactor auro-backtotop to align with the latest auro-button API by replacing the secondary attribute with variant, adding shape/size controls, updating styling imports, bumping dependencies, and updating documentation and demos.

New Features:
- Replace the secondary boolean with a variant attribute to switch between primary and secondary button styles
- Introduce shape and size attributes for dynamically adjusting the button’s appearance on hover and focus

Enhancements:
- Refactor styling to import design tokens via SCSSVariables using @use and streamline host positioning
- Update component API to use the latest auro-button versioning and generate a custom button tag

Build:
- Bump dependencies and devDependencies to the latest @aurodesignsystem and related package versions

Documentation:
- Refresh API documentation, examples, and README to reflect new attribute changes and updated file paths

Chores:
- Add auro-button script imports to demo HTML pages

jason-capsule42 committed:
feat: refactor to use the latest auro-button

BREAKING CHANGE: API refactored to match the latest auro-button version

rmenner committed:
chore: use biome and auro-config

- Added @aurodesignsystem/auro-config.
- Added new configs pointing to auro-config.
- Replaced esLint and scssLint scripts with a unified lint command using biome and stylelint.

AlaskaAirlines/auro-cli#107

rmenner committed:
chore: apply linting updates

AlaskaAirlines/auro-cli#107

rmenner committed:
feat!: upgrade package tooling to use auro-cli

BREAKING CHANGE:
Package exports have changed. To access the component class add /class to the end of the import. See "exports" in the package.json for more details.

AlaskaAirlines/auro-cli#108
AlaskaAirlines/auro-cli#109
AlaskaAirlines/auro-cli#110
AlaskaAirlines/auro-cli#111

rmenner committed:
ci: update workflow to use new auro-action workflows

Workflows added/updated:
- add-project
- check-commits
- check-pr
- codeql
- dev-demo
- pull-request
- release
- sync-tags

AlaskaAirlines/auro-cli#106

jordanjones243 commented on demo/index.md:
```diff
@@ -6,17 +6,21 @@ This file is generated based on a template fetched from `./docs/partials/index.m
 
-<!-- AURO-GENERATED-CONTENT:START (FILE:src=../docs/partials/description.md) -->
-<!-- The below content is automatically added from ../docs/partials/description.md -->
+<auro-icon-9_1_0 customColor category="interface" name="arrow-up" slot="icon" part="icon"></auro-icon-9_1_0>
```
What is this content?

rmenner replied:
I think this is a typo

jordanjones243 commented on demo/index.md:
```diff
@@ -6,17 +6,21 @@ This file is generated based on a template fetched from `./docs/partials/index.m
-<!-- The below content is automatically added from ../docs/partials/description.md -->
+<auro-icon-9_1_0 customColor category="interface" name="arrow-up" slot="icon" part="icon"></auro-icon-9_1_0>
+<br /><br /><br />
+After icon
```
Same here

rmenner replied:
I think this was a typo

jordanjones243 commented on src/auro-backtotop.js:
```diff
@@ -113,6 +96,44 @@ export class AuroBackToTop extends LitElement {
+        type: Boolean,
+        reflect: true,
+      },
+      hidden: {
```
All properties that aren't public API info should be marked as private with a description or not listed under the `properties` getter.

rmenner replied:
I have no context to update them. Technically out of scope for this work.

jordanjones243 commented on src/auro-backtotop.js:
```diff
@@ -113,6 +96,44 @@ export class AuroBackToTop extends LitElement {
+
+      /**
+       * The variant attribute allows for rendering the button using the primary (default) or secondary styles.
+       * @type { 'primary' | 'secondary' | String }
```
Remove `@type` declaration, `primary` and `secondary` are not types

rmenner replied:
This is how to list the "options" that are available in the docs.

jordanjones243 reviewed (changes requested):
Issues:
- When scrolling upwards, the text no longer appears. Reference docssite for original behavior.
- Some component examples still have `rounded` attribute
- First commit needs the `!` flag to mark the `BREAKING CHANGE`
- `ariaLabel` slot example removed
- `custom-elements.json` not in `.gitignore`

rmenner committed:
chore: addressing pr comments

- removed rounded attribute
- add ariaLabel slot example
- add custom-elements.json to .gitignore

rmenner commented:
> Issues:
> 
> * When scrolling upwards, the text no longer appears. Reference docssite for original behavior.
> * Some component examples still have `rounded` attribute
> * First commit needs the `!` flag to mark the `BREAKING CHANGE`
> * `ariaLabel` slot example removed
> * `custom-elements.json` not in `.gitignore`

@jordanjones243 I have addressed these issues. Need @jason-capsule42 for the other fixes.

- removed rounded attribute
- add ariaLabel slot example
- add custom-elements.json to .gitignore
